test/runtests.jl

```diff
@@ -5,127 +5,130 @@ using PerformanceTestTools, Test
-        A = rand(4, 4)
-        @test (@.. A * y' + x) ≈ (@. A * y' + x)
-        @test (@.. A * transpose(y) + x) ≈ (@. A * transpose(y) + x)
-        Ashim = A[1:1, :]
-        @test_throws ArgumentError (@.. Ashim * y' + x)≈(@. Ashim * y' + x) # test fallback
-        @test (@.. broadcast=true Ashim * y'+x) ≈ (@. Ashim * y' + x) # test fallback
-        Av = view(A, 1, :)
-        @test (@.. Av * y' + A) ≈ (@. Av * y' + A)
-        B = similar(A)
-        @.. B .= A
-        @test B == A
-        @.. A = 3
-        @test all(==(3), A)
-        @test (@.. 2 + 3 * 7 - cos(π)) === 24.0
-        foo(f, x) = f(x)
-        @test (@.. foo(abs2, x)) == abs2.(x)
+    nt = (x=x,)
+    @test (@.. (nt.x * y) + x + y + x * (3, 4, 5, 6) + y + x * (1,) + y + 3) ≈
+          (@. (x * y) + x + y + x * (3, 4, 5, 6) + y + nt.x * (1,) + y + 3)
+    A = rand(4, 4)
+    @test (@.. A * y' + x) ≈ (@. A * y' + x)
+    @test (@.. A * transpose(y) + x) ≈ (@. A * transpose(y) + x)
+    Ashim = A[1:1, :]
+    @test_throws ArgumentError (@.. Ashim * y' + x) ≈ (@. Ashim * y' + x) # test fallback
+    @test (@.. broadcast = true Ashim * y' + x) ≈ (@. Ashim * y' + x) # test fallback
+    Av = view(A, 1, :)
+    @test (@.. Av * y' + A) ≈ (@. Av * y' + A)
+    B = similar(A)
+    @.. B .= A
+    @test B == A
+    @.. A = 3
+    @test all(==(3), A)
+    @test (@.. 2 + 3 * 7 - cos(π)) === 24.0
+    foo(f, x) = f(x)
+    @test (@.. foo(abs2, x)) == abs2.(x)
 
-        a = [1, 2]
-        b = [3, 5]
-        c = 1
-        res = @.. @views a = c * b[1:2]
-        @test res == (@. @views a = c * b[1:2])
-        @test a === (@.. a)
-        @static if VERSION >= v"1.6"
-            var".foo"(a) = a
-            @views @.. a = var".foo".(b[1:2]) .+ $abs2(c)
-            @views @.. thread=true a=var".foo".(b[1:2]) .+ $(abs2(c))
-            @test a == [4, 6]
-        else
-            a = [4, 6]
-        end
-        @test (@.. x[2:3, 1] + a) == [6, 9]
-        r = copy(y)
-        @test (@.. r[3:4] += x[2:3, 1] + a) == [13, 17]
-        @test (@views @.. r[3:4] += x[2:3, 1] + a) == [19, 26]
-        @test (@.. @views r[3:4] += x[2:3, 1] + a) == [25, 35]
-        @test r == [5, 6, 25, 35]
-        @test (@.. r + r[end]) == [40, 41, 60, 70]
+    a = [1, 2]
+    b = [3, 5]
+    c = 1
+    res = @.. @views a = c * b[1:2]
+    @test res == (@. @views a = c * b[1:2])
+    @test a === (@.. a)
+    @static if VERSION >= v"1.6"
+      var".foo"(a) = a
+      @views @.. a = var".foo".(b[1:2]) .+ $abs2(c)
+      @views @.. thread = true a = var".foo".(b[1:2]) .+ $(abs2(c))
+      @test a == [4, 6]
+    else
+      a = [4, 6]
+    end
+    @test (@.. x[2:3, 1] + a) == [6, 9]
+    r = copy(y)
+    @test (@.. r[3:4] += x[2:3, 1] + a) == [13, 17]
+    @test (@views @.. r[3:4] += x[2:3, 1] + a) == [19, 26]
+    @test (@.. @views r[3:4] += x[2:3, 1] + a) == [25, 35]
+    @test r == [5, 6, 25, 35]
+    @test (@.. r + r[end]) == [40, 41, 60, 70]
+
+    @.. x = y
+    @test x == y
```

Comment:
@ChrisRackauckas https://github.com/SciML/RecursiveArrayTools.jl/actions/runs/9298202625/job/25589645929?pr=382#step:6:1452 should now pass.